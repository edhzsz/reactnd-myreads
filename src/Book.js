import React from 'react'
import BookShelfChanger from './BookShelfChanger.js'

let Book = ({ name, authors, coverWidth, coverHeight, coverUrl}) => (
    <div className="book">
      <div className="book-top">
        <div className="book-cover" style={{ width: coverWidth, height: coverHeight, backgroundImage: `url("${coverUrl}")` }}></div>
          <BookShelfChanger />
        </div>
      <div className="book-title">{name}</div>
      <div className="book-authors">{authors}</div>
    </div>
  );

export default Book;