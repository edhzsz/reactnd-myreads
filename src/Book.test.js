import React from 'react';
import ReactDOM from 'react-dom';
import Book from './Book.js';
import renderer from 'react-test-renderer';

describe('Component: Book', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Book />, div);
  });

  it('should match its empty snapshot', () => {
    const tree = renderer.create(
      <Book />
     ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render the props', () => {
    const tree = renderer.create(
      <Book name="Test Name" authors="test authors" coverWidth="100" coverHeight="200" coverUrl="test-url" />
     ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});