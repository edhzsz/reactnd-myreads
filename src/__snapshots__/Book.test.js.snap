// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Book should match its empty snapshot 1`] = `
<div
  className="book"
>
  <div
    className="book-top"
  >
    <div
      className="book-cover"
      style={
        Object {
          "backgroundImage": "url(\\"undefined\\")",
          "height": undefined,
          "width": undefined,
        }
      }
    />
    <div
      className="book-shelf-changer"
    >
      <select>
        <option
          disabled={true}
          value="none"
        >
          Move to...
        </option>
        <option
          value="currentlyReading"
        >
          Currently Reading
        </option>
        <option
          value="wantToRead"
        >
          Want to Read
        </option>
        <option
          value="read"
        >
          Read
        </option>
        <option
          value="none"
        >
          None
        </option>
      </select>
    </div>
  </div>
  <div
    className="book-title"
  />
  <div
    className="book-authors"
  />
</div>
`;

exports[`Component: Book should render the props 1`] = `
<div
  className="book"
>
  <div
    className="book-top"
  >
    <div
      className="book-cover"
      style={
        Object {
          "backgroundImage": "url(\\"test-url\\")",
          "height": "200",
          "width": "100",
        }
      }
    />
    <div
      className="book-shelf-changer"
    >
      <select>
        <option
          disabled={true}
          value="none"
        >
          Move to...
        </option>
        <option
          value="currentlyReading"
        >
          Currently Reading
        </option>
        <option
          value="wantToRead"
        >
          Want to Read
        </option>
        <option
          value="read"
        >
          Read
        </option>
        <option
          value="none"
        >
          None
        </option>
      </select>
    </div>
  </div>
  <div
    className="book-title"
  >
    Test Name
  </div>
  <div
    className="book-authors"
  >
    test authors
  </div>
</div>
`;
